internal/formats/common/cyclonedxhelpers/component.go

```diff
@@ -25,6 +25,11 @@ func encodeComponent(p pkg.Package) cyclonedx.Component {
 		properties = &props
 	}
 
+	bomRef := p.PURL
```

Comment:
Maybe `p.ID()` would be a better thing to use here? You should not have to call `p.SetID()`, as it is called if for some reason a package is added to the catalog without having an ID.

Reply:
The CycloneDX docs prefer PURL as the bom ref https://cyclonedx.org/use-cases/#dependency-graph

Reply:
Here's my concern: PURLs are not unique within the Syft data model. For example, if we scan `package-lock.json` and _also_ scan `node_modules` that has been populated, we will find 2 packages with the same PURL but they will not be unique in Syft because of the location of the evidence, whereas the `p.ID()` will be unique.

Reply:
I wouldn't prefer using syft ids as bom refs since having readable bom refs also helps with the VEX readability. Would an alternative like trying to dedupe and merge components with same ids so that we preserve the location/source they were found in, in an array but have a common top level component?

Reply:
I, personally, like this quite a bit: if we find evidence in multiple locations pointing to the same package, this should be reflected in `locations` rather than multiple packages.